src/gen_bpf.c

```diff
@@ -1532,11 +1576,17 @@ static int _gen_bpf_syscalls(struct bpf_state *state,
 		_sys_sort(db_secondary->syscalls, &s_head, &s_tail, optimize);
 
 	if (optimize == 2) {
+		syscall_cnt = _get_syscall_cnt(state, s_tail);
 		rc = _gen_bpf_init_bintree(&bintree_hashes, &bintree_syscalls,
-					   bintree_levels, db->syscall_cnt,
+					   bintree_levels, syscall_cnt,
 					   &empty_cnt);
 		if (rc < 0)
 			goto out;
+
+		/* reset the syscall_cnt variable because later in this
+		 * function it's used as a counter
+		 */
+		syscall_cnt = 0;
```

Comment:
We can drop this special handling if we move the `syscall_cnt` initialization to 0 from the declaration line to right about the for-loop where it is first used.